docs/development/local-plugin-creation.md

```diff
@@ -0,0 +1,292 @@
+
+if [ ! -d "$DUPLICATE_ENVIRONMENT_ROOT_OPERATIONS" ]; then
+  echo "No duplicate-environment directory.  Skipping."
+  exit 0
+fi
+
+printf "Deploying duplicate-environment..."
+
+if [ ! -f "$BITOPS_SCHEMA_ENV_FILE" ]; then 
+  echo "No duplicate-environment ENV file found"
+else
+  source "$BITOPS_SCHEMA_ENV_FILE"
+fi
+
+bash $SCRIPTS_DIR/validate_env.sh
+
+cd $DUPLICATE_ENVIRONMENT_ROOT_OPERATIONS
+
+echo "Listing contents of duplicate-environment Root: $DUPLICATE_ENVIRONMENT_ROOT_OPERATIONS"
+ls -al .
+
+echo "settings:"
+echo "DUPLICATE_ENVIRONMENT_FOO"
+echo "$DUPLICATE_ENVIRONMENT_FOO"
+```
+> **Note:** Much of the above is best practice boilerplate and is not strictly necessary.
+
+`scripts/validate_env.sh`
+```
+#!/bin/bash
+set -e 
+
+if [ -z "$BITOPS_ENVIRONMENT" ]; then
+  echo "environment variable (ENVIRONMENT) not set"
+  exit 1
+fi
+```
+
+`plugin.config.yaml`
+```
+plugin:
+  deployment:
+    language: bash
+    deployment_script: deploy.sh
+    core_schema_parsing: true
+    life_cycle_scripts: true
+```
+
+
+## 3. Create an ops repo for testing
+```
+cd /path/to/ops-repos
+mkdir duplicate-environment
+cd duplicate-environment
+```
+
+From there, you'll need to create an environment with a directory that matches your tool.
+
+`mkdir -p /path/to/ops-repo/plugin-no-install/duplicate-environment`
+
+Populate the tool's `bitops.config.yaml` based on the schema defined above:
+`/path/to/ops-repo/plugin-no-install/duplicate-environment/bitops.config.yaml`
+```
+duplicate-environment:
+  properties:
+    foo: baz
+```
+
+## 4. Test your plugin
+
+### 4.1. BitOps-level BitOps Config
+To test your plugin, you'll need BitOps to run with a BitOps-level BitOps config that has your plugin defined in the `deployments`.
+
+Create a `bitops.config.yaml` somewhere (say: `/path/to/ops-repo/plugin-no-install/duplicate-environment/bitops-level/bitops.config.yaml`), and add a `plugins` and `deployments` reference to your plugin:
+```
+bitops:
+  fail_fast: true
+  logging:      
+    level: DEBUG              # Sets the logging level
+    color:
+      enabled: true           # Enables colored logs
+  plugins:    
+    duplicate-environment: {}
+  deployments:
+    duplicate-environment:
+      plugin: duplicate-environment
+```
+> **NOTE:** `plugins.duplicate-environment` is empty because it's only used as a reference for `deployments.duplicate-environment`.
+
+### 4.2. Run your test
+To run BitOps against a local plugin, you'll need to mount the plugin to the location BitOps expects plugins to be.
+```
+docker run --rm --name bitops \
+-e BITOPS_ENVIRONMENT="duplicate-environment" \
+-e PYTHONUNBUFFERED=1 \
+-v /path/to/bitops:/opt/bitops \
+-v /path/to/ops-repo/plugin-no-install:/opt/bitops_deployment \
+-v /path/to/bitops-level/bitops.config.yaml:/opt/bitops/bitops.config.yaml \
+-v /opt/bitops/scripts/plugins/terraform \
+-v /path/to/bitops-plugins/duplicate-environment:/opt/bitops/scripts/plugins/duplicate-environment \
+bitovi/bitops:2.1.0
```

Comment:
For development I'd recommend using the `dev` tag, it will refer to the latest unreleased code from the `main` branch, instead of released one.

```suggestion
bitovi/bitops:dev
```

the was exactly the idea behind it